ui/selectors/transactions.js

```diff
@@ -36,6 +36,19 @@ const allowedSwapsSmartTransactionStatusesForActivityList = [
 
 export const unapprovedMsgsSelector = (state) => state.metamask.unapprovedMsgs;
 
+export const getTransactions = createDeepEqualSelector(
+  (state) => {
+    const { transactions } = state.metamask ?? {};
+
+    if (!transactions?.length) {
+      return [];
+    }
+
+    return transactions.sort((a, b) => a.time - b.time); // Ascending
+  },
+  (transactions) => transactions,
+);
+
 export const getCurrentNetworkTransactions = createDeepEqualSelector(
```

Comment:
Should we change this to use `getTransactions` as a dependency to avoid the duplicated sorting?

Reply:
I tried that yesterday but some of the unit tests started to fail. After your comment, I tried again, found the issue in the unit tests, and pushed this change to use `getTransactions`.